### DISCUSSION TITLE
Java client doesn't show as service in Jaeger UI

### DISCUSSION THREAD

**Discussion by stevensim226** [jaegertracing/jaeger#2908] [Q&A]
Hello,
I am currently trying to integrate my existing java web app (I am not using springboot) w/ Jaeger in the **kubernetes cluster**

I have already succeeded in deploying Jaeger UI and the sidecar pod but the problem is that it seems Jaeger UI doesn't show my java app as a service, w/ that I can't see any metrics from it.

For my kubernetes manifests for jaeger, I'm using as what is already available in here
CRD:
https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/crds/jaegertracing.io_jaegers_crd.yaml

RBAC:
https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/service_account.yaml
https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/role.yaml
https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/role_binding.yaml

Jaeger Operator:
https://raw.githubusercontent.com/jaegertracing/jaeger-operator/master/deploy/operator.yaml

For my java settings, I followed [this documentation](https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/README.md) and [this official sample](https://github.com/yurishkuro/opentracing-tutorial/blob/master/java/src/main/java/lesson01/solution/Hello.java) and added these line of code into my `main` function for debugging
```
System.out.println("Initializing Jaeger's Open Tracer... 123");
Tracer tracer = new JaegerTracer.Builder("someservicename").build();

Span span = tracer.buildSpan("say-hello").start();
span.setTag("sampletag", "samplevalue");

String helloStr = "samplehello";
span.log(ImmutableMap.of("event", "string-format", "value", helloStr));

System.out.println(helloStr);
span.log(ImmutableMap.of("event", "println"));

System.out.println("Finished initializing Jaeger's Open Tracer 123");
```

But it doesn't show `"someservicename"` in the jaeger UI in here
<img width="997" alt="Screen Shot 2021-03-27 at 01 16 52" src="https://user-images.githubusercontent.com/57280118/112675749-1dbd8700-8e9a-11eb-91db-305b7cfee83c.png">

FYI: my app is running on port 8087 , and I have not added any environment variables

I have also followed troubleshooting guide in https://www.jaegertracing.io/docs/1.17/troubleshooting/#check-the-metrics-endpoint
but the `/metrics` endpoint didn't return any of the 4 key:value pairs (`jaeger_collector_spans_received`, etc.)

**Comment by joe-elliott** [comment]
If those metrics are absent it means that the collector is not receiving any traces. I would check your agent metrics next to see if they are receiving spans and forwarding them correctly.

This guide walks through each component and metrics you can check to make sure that spans are forwarding correctly: https://medium.com/jaegertracing/where-did-all-my-spans-go-a-guide-to-diagnosing-dropped-spans-in-jaeger-10d9697f8182

**Reply by stevensim226** [reply]
Hello @joe-elliott , thanks for the answer,
I have read your article and I think the problem is as what you said that the Jaeger Collector not receiving any traces since I can't find anything related to `jaeger_collector` from Prometheus (I did find things that are related to Agent, Inprocess Client, etc.)

Although I can't seem to figure out the way to fix it, is there a port that needs to be exposed, env vars needed, or maybe my Java code itself is wrong?

**Comment by albertteoh** [comment]
Were you able to ascertain whether jaeger-agent was receiving spans from
your application from those metrics?

You can find the ports that need to be exposed by the collector here:
https://www.jaegertracing.io/docs/1.22/deployment/#collectors.

Typically, the agent sends spans to jaeger-collector's port *14250* (gRPC).

Albert

**Comment by stevensim226** [comment]
Case resolved, turns out I only forgot to call span.finish() on my code.

a quite useful article about java example that I used is this https://www.scalyr.com/blog/jaeger-tracing-tutorial/

@albertteoh thanks for the help anyways, it has been opened as a k8s service.
